core/src/main/scala/shapeless/records.scala

```diff
@@ -98,6 +99,13 @@ class RecordMacros(val c: whitebox.Context) {
   val SymTpe = typeOf[scala.Symbol]
   val atatTpe = typeOf[tag.@@[_,_]].typeConstructor
 
+  def mkRecordEmptyImpl(method: Tree)(rec: Tree*): Tree = {
+    if(rec.nonEmpty)
+      c.abort(c.enclosingPosition, "this method must be called with named arguments")
```

Comment:
This error message appears when `Record` is called like
```scala
Record(2, "a")
```

So this error message ("must be called with *named* arguments") looked like the right one to me.